modules/public/logic.py

```diff
@@ -0,0 +1,258 @@
+
+    return users_by_country
+
+
+def build_country_info(users_by_country, display_number_users=False):
+    """Build info list for every country
+
+    :param users_by_country: number of users per country
+    :param display_number_users: Add the number of users if True
+
+    :returns: A dictionnary with the country information
+    """
+    country_data = {}
+    for country in pycountry.countries:
+        country_info = users_by_country.get(country.alpha_2)
+        number_of_users = 0
+        percentage_of_users = 0
+        color_rgb = [247, 247, 247]
+        if country_info is not None:
+            if display_number_users:
+                number_of_users = country_info['number_of_users'] or 0
+            percentage_of_users = country_info['percentage_of_users'] or 0
+            color_rgb = country_info['color_rgb'] or [247, 247, 247]
+
+        # Use common_name if available to be less political offending (#310)
+        try:
+            country_name = country.common_name
+        except AttributeError:
+            country_name = country.name
+
+        country_data[country.numeric] = {
+            'name': country_name,
+            'code': country.alpha_2,
+            'percentage_of_users': percentage_of_users,
+            'color_rgb': color_rgb
+        }
+
+        if display_number_users:
+            country_data[country.numeric]['number_of_users'] = number_of_users
+
+    return country_data
+
+
+def get_searched_snaps(search_results):
+    """Get search snaps from API response
+
+    :param search_results: the body responsed by the API
+
+    :returns: The list of the searched snaps
+    """
+    return (
+        search_results['_embedded']['clickindex:package']
+        if '_embedded' in search_results
+        else []
+    )
+
+
+def get_pages_details(url, links):
+    """Transform returned navigation links from search API from limit/offset
+    to size/page
+
+    :param url: The url to build
+    :param links: The links returned by the API
+
+    :returns: A dictionnary with all the navigation links
+    """
+    links_result = {}
+
+    if('first' in links):
+        links_result['first'] = convert_limit_offset_to_size_page(
+            url,
+            links['first']['href']
+        )
+
+    if('last' in links):
+        links_result['last'] = convert_limit_offset_to_size_page(
+            url,
+            links['last']['href']
+        )
+
+    if('next' in links):
+        links_result['next'] = convert_limit_offset_to_size_page(
+            url,
+            links['next']['href']
+        )
+
+    if('prev' in links):
+        links_result['prev'] = convert_limit_offset_to_size_page(
+            url,
+            links['prev']['href']
+        )
+
+    if('self' in links):
+        links_result['self'] = convert_limit_offset_to_size_page(
+            url,
+            links['self']['href']
+        )
+
+    return links_result
+
+
+def convert_limit_offset_to_size_page(url, link):
```

Comment:
I think it would make sense to call this `convert_navigation_url`, since it takes and returns a whole URL.